agent/statemachine/state.go

```diff
@@ -89,6 +90,20 @@ func (sm *stateMachine) Start(ctx context.Context) error {
 	}
 }
 
+func (sm *stateMachine) Reset(initialState State) {
+	sm.mu.Lock()
+	defer sm.mu.Unlock()
+
+	// Reset current state to initial state
+	sm.currentState = initialState
+
+	// Close the existing event channel to stop processing events
+	close(sm.eventChan)
```

Comment:
Is this safe? Do we have a select on the consumer routine to catch this?

Reply:
this happens after the consumer is stopped, we cancel context first so this is safe